File: array/0268_missing_number/0268_missing_number.py
# -*- coding: utf-8 -*-


class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        import operator
        return reduce(operator.xor, nums, \
                      reduce(operator.xor, xrange(len(nums) + 1)))


print(Solution().missingNumber([1,2,3,4,5,6]))

File: array/0268_missing_number/0268_missing_number.go
package main

import (
	"fmt"
)

func missingNumber(nums []int) int {
	xor := 0

	for i, n := range nums {
		xor ^= i ^ n
	}

	return xor ^ len(nums)
}

func main() {
	nums := []int{3, 0, 1}
	fmt.Println(missingNumber(nums))
}
